### ISSUE TITLE
Something went wrong with LiveReactload initialization

### ISSUE THREAD

**Issue by derekr** [milankinen/livereactload#18]
Thanks for implementing this! Super excited to get it up and running.

Getting the following warning multiple times on page load:
`Something went wrong with LiveReactload initialization. Reloading is not enabled.`

![screen shot 2015-03-31 at 11 31 26 am](https://cloud.githubusercontent.com/assets/22249/6926190/7d989912-d799-11e4-8c25-23bcc7199163.png)

Not quite sure how to debug. Here is my watch setup:

```makefile
#
# Development and watch tasks.
#
dev: setup public/build/js/vendor.bundle.js
	@node_modules/.bin/parallelshell \
		'make watch-js' \
		'make watch-less' \
		'make livereactload-listen' \
		'make livereactload-notify' \
		'make live-reload' \
		'NODE_ENV=development node index.js | node_modules/.bin/garnish'

watch-less:
	@node_modules/.bin/nodemon \
		--quiet \
		--ext less \
		--watch client \
		--exec 'make less' \
		| node_modules/.bin/garnish

public/build/js/vendor.bundle.js:
	@node_modules/.bin/browserify \
		-r react \
		-r react/addons \
		-r react-router \
		-g uglifyify \
		-o public/build/js/vendor.bundle.js

watch-js:
	@node_modules/.bin/watchify client/main.jsx \
		-x react \
		-x react/addons \
		-x react-router \
		-t reactify \
		-t envify \
		-g livereactload \
		-o public/build/js/main.bundle.js \
		-dv

livereactload-notify:
	node_modules/.bin/nodemon \
		--quiet \
		--watch public/build/js \
		--ext js \
		--exec 'node_modules/.bin/livereactload notify'

livereactload-listen:
	node_modules/.bin/livereactload listen

live-reload:
	@node_modules/.bin/wtch --dir public --extension css \
	 | node_modules/.bin/garnish
```

And the output after running `make dev` and changing a component:

```sh
life-events → make dev
node_modules/.bin/livereactload listen
node_modules/.bin/nodemon \
		--quiet \
		--watch public/build/js \
		--ext js \
		--exec 'node_modules/.bin/livereactload notify'
LiveReactload: server listening in port 4474
 info wtch: livereload running on 35729
 info app: started -> http://0.0.0.0:8080
LiveReactload: sending reload request to 0 clients
879136 bytes written to public/build/js/main.bundle.js (1.42 seconds)
LiveReactload: sending reload request to 0 clients
LiveReactload: new client connected
879141 bytes written to public/build/js/main.bundle.js (0.18 seconds)
LiveReactload: sending reload request to 1 clients
```

Everything looks like it's working, but no reloads are happening. I have caching disabled on the client so not sure if it's related to that. Anyway any insight is appreciated!

**Comment by milankinen** [comment]
I have two guesses:

1) Your uglify is somehow eliminating dead code and removes the reactload initialization logic
2) You are using react as an external dependency right? That might also cause some issues

Can you show your `bundle.js` so that I can investigate its contents?

**Comment by derekr** [comment]
Here here is a gist w/ out the uglify transform. I did a quick test w/ bundling everything in to a single file, but still no luck.

https://gist.github.com/derekr/4492240600e4ec6d618d

**Comment by milankinen** [comment]
Hmmm.. complains about missing `react` for me. This seems very strange because that warning should never occur... Is it possible to produce an example project containing this problem so that I could clone it and take a closer look at it?

**Comment by derekr** [comment]
Yeah for sure. Will try to get one up this afternoon.

**Comment by milankinen** [comment]
I think I got it now. Accidentally tried to inject LiveReactload to transient dependencies (which were required during the initialization). Fix coming: https://github.com/milankinen/livereactload/commit/fcd6929b88981c578f6ef3d3b6cb370fc09247a5

**Comment by milankinen** [comment]
Should be fixed in `0.4.0`. I'll open this if problem still exists
